### PR TITLE
[simple-hub] deploy optimization

### PR DESCRIPTION
**Pull request by kerzhner**
Anecdotal evidence: when docker container does not need rebuilding (cache is fully used), the job takes 6m20s. When docker container needs to be rebuilt from scratch (cache is not used), the job duration is 11m30s. Before this PR, the prepare + hub release jobs took 13m30s.

- Standalone `release-hub` job that does not require `prepare`. Saves time by removing workspace persistence and restoration that is required when the release process is 2 jobs.
- Use docker caching.
- Minimize content of docker context.
- Add release instructions to hub readme.
- `rm -rf` all packages not needed by the hub to speed up `yarn`.
- Use `yarn.lock` in the hub docker container to ensure that the same dependency versions are used as during development. 

Closes https://github.com/statechannels/monorepo/issues/1583.

### CODE REVIEW

**Comment by kerzhner** [comment]
Fixes https://github.com/statechannels/monorepo/issues/1583

File: .circleci/config.yml [hunk 1]
```diff
@@ -309,6 +285,32 @@ jobs:
...
+    working_directory: /home/circleci/project
+    resource_class: large
+    docker:
+      - image: circleci/node:10.16
+    steps:
+      - setup_remote_docker
+      - checkout
+      - <<: *restore_dep
+      - <<: *restore_lib
+      - run: rm -rf app-wallet-interface channel-provider channel-client \
```
[the remark this answers is not in the document]
Comment (kerzhner) [reply .circleci/config.yml:298]: 😁

File: packages/simple-hub/docker/simple-hub.dockerfile.dockerignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# A .dockerignore symlink to this file exists in the monorepo root
+# circleci uses a docker version that does not support buildkit.
+# .dockerignore must be located at the root of the context and have a generic name.
+# https://github.com/statechannels/monorepo/issues/1615 tracks the buildkit issue.
+
+# Ignore all first, then add back directories that are needed.
+
```
Comment (kerzhner) [inline packages/simple-hub/docker/simple-hub.dockerfile.dockerignore:7]:
```suggestion
```

File: packages/simple-hub/tsconfig.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
     "module": "commonjs",
     "target": "es2017",
     "noImplicitAny": false,
+    "rootDir": ".",
```
Comment (snario) [inline packages/simple-hub/tsconfig.json:8]: What made this necessary? Just curious

Comment (kerzhner) [reply packages/simple-hub/tsconfig.json:8]: Typescript compilation started failing for me locally (again...). And I had to add this to get rid of the errors for importing `env` module.

**Pull request merged**